pyqtgraph/parametertree/parameterTypes.py

```diff
@@ -681,3 +682,241 @@ class TextParameter(Parameter):
+    itemClass = ProgressBarParameterItem
+
+registerParameterType('progress', ProgressBarParameter, override=True)
+
+class SliderParameterItem(WidgetParameterItem):
+    def makeWidget(self):
+        w = QtGui.QSlider()
+        w.setOrientation(1)
+        w.setMaximumHeight(20)
+        w.sigChanged = w.valueChanged
+        def setLimits(limits):
+            w.setMinimum(min(limits))
+            w.setMaximum(max(limits))
+        self.limitsChanged = setLimits
+        self.limitsChanged(self.param.opts['limits'])
+        self.widget = w
+        self.hideWidget = False
+        return w
+
+class SliderParameter(Parameter):
+    itemClass = SliderParameterItem
+
+registerParameterType('slider', SliderParameter, override=True)
+
+class FontParameterItem(WidgetParameterItem):
+    def makeWidget(self):
+        w = QtGui.QFontComboBox()
+        w.setMaximumHeight(20)
+        w.sigChanged = w.currentFontChanged
+        w.value = w.currentFont
+        w.setValue = w.setCurrentFont
+        self.widget = w
+        self.hideWidget = False
+        return w
+
+class FontParameter(Parameter):
+    itemClass = FontParameterItem
+
+registerParameterType('font', FontParameter, override=True)
+
+class CalendarParameterItem(WidgetParameterItem):
+    def makeWidget(self):
+        w = QtGui.QCalendarWidget()
+        w.setMaximumHeight(200)
+        w.sigChanged = w.selectionChanged
+        w.value = w.selectedDate
+        w.setValue = w.setSelectedDate
+        self.widget = w
+        self.hideWidget = False
+        self.param.opts['default'] = QtCore.QDate.currentDate()
+        return w
+
+class CalendarParameter(Parameter):
+    itemClass = CalendarParameterItem
+
+registerParameterType('calendar', CalendarParameter, override=True)
+
+
+
+class PenParameterItem(ParameterItem):
+    def __init__(self, param, depth):
+        ParameterItem.__init__(self, param, depth)
+        self.layoutWidget = QtGui.QWidget()
+        self.layout = QtGui.QHBoxLayout()
+        self.layoutWidget.setLayout(self.layout)
+        self.button = QtGui.QPushButton()
+        #larger button
+        self.button.setFixedWidth(100)
+        self.layout.addWidget(self.button)
+        self.layout.addStretch()
+        self.pen = QtGui.QPen()
+        self.button.clicked.connect(self.buttonClicked)
+        #clear button for drawing, override paint event
+        self.setText(0, '')
+        self.button.paintEvent = self.buttonPaintEvent
+        
+
+    def value(self):
+        return self.pen
+
+    def setValue(self,pen):
+        self.penChanged(pen)
+
+    def treeWidgetChanged(self):
+        ParameterItem.treeWidgetChanged(self)
+        tree = self.treeWidget()
+        if tree is None:
+            return
+
+        tree.setFirstItemColumnSpanned(self, True)
+        tree.setItemWidget(self, 0, self.layoutWidget)
+
+    def buttonClicked(self):
+        #open up the pen selector dialog
+        self.oldPen = fn.mkPen(self.pen)
```

Comment:
This gets the current value from the parameter tree
```suggestion
        self.oldPen = fn.mkPen(self.param.value())
```